src/controller/CHIPDevice.h

```diff
@@ -344,6 +345,15 @@ class DLL_EXPORT Device : public Messaging::ExchangeDelegate, public SessionEsta
 
     CASESession & GetCASESession() { return mCASESession; }
 
+    CHIP_ERROR GenerateCSRNonce(ByteSpan & nonce)
```

Comment:
How about not taking an argument at all? It seems one can use "GetCSRNonce" to actually get it.